File: bridge/java/BridgeClient.java
class Abstraction {
  Implementor implementor;
  void operation() {
    implementor.operationImp();
  }
}

class Implementor {
  void operationImp() {
    System.out.println("Implementor instance has been created");
  }
}

class RefinedAbstraction extends Abstraction {
  public RefinedAbstraction(String classname) {

    try {
      implementor = (Implementor) Class.forName(classname).newInstance();
    } catch (Exception e) {
      e.printStackTrace();
    }
  }
}

class ConcreteImplementorA extends Implementor {
  @Override
  void operationImp() {
    System.out.println("ConcreteImplementorA instance has been created");
  }
}

class ConcreteImplementorB extends Implementor {
  @Override
  void operationImp() {
    System.out.println("ConcreteImplementorB instance has been created");
  }
}

public class BridgeClient {
  public static void main(String[] args) {
    Abstraction abstraction = new RefinedAbstraction(ConcreteImplementorA.class.toString().substring(6));
    abstraction.operation();
    abstraction = new RefinedAbstraction((ConcreteImplementorB.class.toString().substring(6)));
    abstraction.operation();
  }
}

File: bridge/python/BridgeClient.py
#!/usr/bin/env python
# coding:utf8
# @Date    : 2018/7/22
# @Author  : Koon
# @Link    : koon.cool
# When I wrote this, only God and I understood what I was doing. Now, God only knows.

'''
    ┌─┐       ┌─┐
 ┌──┘ ┴───────┘ ┴──┐
 │                 │
 │       ───       │
 │  ─┬┘       └┬─  │
 │                 │
 │       ─┴─       │
 │                 │
 └───┐         ┌───┘
     │         │
     │         │
     │         │
     │         └──────────────┐
     │                        │
     │                        ├─┐
     │                        ┌─┘
     │                        │
     └─┐  ┐  ┌───────┬──┐  ┌──┘
       │ ─┤ ─┤       │ ─┤ ─┤
       └──┴──┘       └──┴──┘
           神兽保佑
           永无BUG!
'''
class Implementor(object):
  def operationImp(self):
    print('Implementor instance has been created')

class Abstraction(object):
  def operation(self):
    self._imp.operationImp()

class ConcreteImplementorA(Implementor):
  def operationImp(self):
    print('ConcreteImplementorA instance has been created')

class ConcreteImplementorB(Implementor):
  def operationImp(self):
    print('ConcreteImplementorB instance has been created')

class RefinedAbstraction(Abstraction):
  def __init__(self, cls):
    self._imp = cls()

if __name__ == '__main__':
  abstraction = RefinedAbstraction(ConcreteImplementorA)
  abstraction.operation()
  abstraction = RefinedAbstraction(ConcreteImplementorB)
  abstraction.operation()